src/util.ts

```diff
@@ -7,3 +7,9 @@ export function convertPkcs1ToPkcs8(pkcs1: string): string {
   const key = new NodeRSA(pkcs1, 'pkcs1')
   return key.exportKey('pkcs8')
 }
+
+/**
+ * Regex pattern that the code verifier and code challenge in the PKCE flow should match according to the PKCE RFC
+ * https://www.rfc-editor.org/rfc/rfc7636
+ */
+export const codeVerifierAndChallengePattern = /^[A-Za-z\d\-._~]{43,128}$/
```

Comment:
Suggestion: Should we put this under `constants.ts` instead? Especially when we are not using this regex pattern directly in this file anyways.

Reply:
yup got it

Reply:
Refactored in this [PR](https://github.com/opengovsg/sgid-client/pull/39) in order to prevent this PR from growing too large